tests/wpt_tests.cpp

```diff
@@ -154,7 +169,7 @@ bool setters_tests_encoding() {
 
       if (category == "protocol") {
         std::string_view expected = element["expected"]["protocol"];
-        ada::set_scheme(base, std::string{new_value});
+        ada::set_scheme(base, std::string{new_value}, type);
```

Comment:
Same applies to here regarding the encoding type parameter.

Reply:
It is a parameter in the API: https://github.com/ada-url/ada/blob/f0298674750eeae848c5859aff062e91ef248e4c/include/ada/implementation.h#L44

We must either remove it from the API or include tests for it.